src/sdk/namespace_blob.js

```diff
@@ -409,16 +411,39 @@ class NamespaceBlob {
         return { obj_id };
     }
 
+    // Adhere to azcopy's ResolveInvalidKey for handling of invalid metadata keys
+    _rename_invalid_md_key(xattr, md_key) {
+        const valid_key = md_key.replace(invalid_azure_md_regex, '_');
+        xattr[XATTR_RENAME_PREFIX + valid_key] = xattr[md_key];
+        xattr[XATTR_RENAME_KEY_PREFIX + valid_key] = md_key;
+        delete xattr[md_key];
+    }
+
+    _exclude_invalid_md_key(xattr, md_key) {
+            delete xattr[md_key];
+    }
+
     _check_valid_xattr(params) {
-        // This md validation check is a part of namespace blob because but Azure Blob 
-        // accepts C# identifiers only but S3 accepts other xattr too.
-        const is_invalid_attr = ([key, val]) => !valid_attr_regex.test(key);
-        const invalid_attr = Object.entries(params.xattr).find(is_invalid_attr);
-        if (invalid_attr) {
-            const err = new Error('InvalidMetadata: metadata keys are invalid.');
-            err.rpc_code = 'INVALID_REQUEST';
-            throw err;
-        }
+        const modified_xattr = _.cloneDeep(params.xattr);
+        Object.keys(params.xattr).forEach(md_key => {
```

Comment:
I think that using `for of` is a better style than `forEach`. it is more readable, and also support standard control flow keywords (`break`, `continue`, etc.)

```suggestion
        for (const md_key of Object.keys(params.xattr)) {
```